nexus/src/external_api/http_entrypoints.rs

```diff
@@ -2237,6 +2264,416 @@ async fn certificate_delete(
+}
+
+/// List the blocks in an address lot.
+#[endpoint {
+    method = GET,
+    path = "/v1/system/networking/address-lot/{address_lot}/blocks",
+    tags = ["external-networking"],
+}]
+async fn networking_address_lot_block_list(
+    rqctx: RequestContext<Arc<ServerContext>>,
+    path_params: Path<params::AddressLotPath>,
+    query_params: Query<PaginatedById>,
+) -> Result<HttpResponseOk<ResultsPage<AddressLotBlock>>, HttpError> {
+    let apictx = rqctx.context();
+    let handler = async {
+        let nexus = &apictx.nexus;
+        let query = query_params.into_inner();
+        let path = path_params.into_inner();
+        let pagparams = data_page_params_for(&rqctx, &query)?;
+        let opctx = crate::context::op_context_for_external_api(&rqctx).await?;
+        let address_lot_lookup =
+            nexus.address_lot_lookup(&opctx, path.address_lot)?;
+        let blocks = nexus
+            .address_lot_block_list(&opctx, &address_lot_lookup, &pagparams)
+            .await?
+            .into_iter()
+            .map(|p| p.into())
+            .collect();
+
+        Ok(HttpResponseOk(ScanById::results_page(
+            &query,
+            blocks,
+            &|_, x: &AddressLotBlock| x.id,
+        )?))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Create a loopback address.
+#[endpoint {
+    method = POST,
+    path = "/v1/system/networking/loopback-address",
+    tags = ["external-networking"],
+}]
+async fn networking_loopback_address_create(
+    rqctx: RequestContext<Arc<ServerContext>>,
+    new_loopback_address: TypedBody<params::LoopbackAddressCreate>,
+) -> Result<HttpResponseCreated<LoopbackAddress>, HttpError> {
+    let apictx = rqctx.context();
+    let handler = async {
+        let nexus = &apictx.nexus;
+        let params = new_loopback_address.into_inner();
+        let opctx = crate::context::op_context_for_external_api(&rqctx).await?;
+        let result = nexus.loopback_address_create(&opctx, params).await?;
+
+        let addr: LoopbackAddress = result.into();
+
+        Ok(HttpResponseCreated(addr))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+#[derive(Serialize, Deserialize, JsonSchema)]
+pub struct LoopbackAddressPath {
+    /// The rack to use when selecting the loopback address.
+    pub rack_id: Uuid,
+
+    /// The switch location to use when selecting the loopback address.
+    pub switch_location: Name,
+
+    /// The IP address and subnet mask to use when selecting the loopback
+    /// address.
+    pub address: IpAddr,
+
+    /// The IP address and subnet mask to use when selecting the loopback
+    /// address.
+    pub subnet_mask: u8,
+}
+
+/// Delete a loopback address.
+#[endpoint {
+    method = DELETE,
+    path = "/v1/system/networking/loopback-address/{rack_id}/{switch_location}/{address}/{subnet_mask}",
```

Comment:
I'm really not sure about this API. There's probably a follow up here that once we merge this PR and unify switch-ports/switches then we should probably have this take a switch selector. The address/subnet_mask part _may_ be fine as is? I'm not sure, it's strange to my eye.

Reply:
Ha, I just realized that because the subnet mask is usually represented by a `/` this makes it look like an extra path param in the route. @internet-diglett does this _always_ require a subnet mask? If not we may need two endpoints here that do essentially the same thing w/ different paths.